server/src/main/java/org/elasticsearch/action/admin/cluster/node/reload/TransportNodesReloadSecureSettingsAction.java

```diff
@@ -78,15 +82,39 @@ protected NodeRequest newNodeRequest(NodesReloadSecureSettingsRequest request) {
         return new NodesReloadSecureSettingsResponse.NodeResponse();
     }
 
+    @Override
+    protected void doExecute(Task task, NodesReloadSecureSettingsRequest request,
+                             ActionListener<NodesReloadSecureSettingsResponse> listener) {
+        if (request.hasPassword() && isNodeLocal(request) == false && isNodeTransportTLSEnabled() == false) {
+            request.closePassword();
+            listener.onFailure(
```

Comment:
Could this checking be done at the rest layer, rather than in the transport action?

Reply:
I had similar concerns when implementing this but I think that it makes more sense to chack this here, especially because of `isNodeTransportTLSEnabled()`. Do you feel strongly otherwise?

Reply:
I would feel more comfortable if we called `executeLocally` on the nodeclient directly, so we are absolutely sure we are no passing this execution off to another node before we have checked whether we are using ssl. But it's a minor point because we've already parsed the request which has the passphrase.